<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1283</width>
    <height>886</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OntoGenix</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QTabWidget" name="PLANNING">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Planning</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QGridLayout" name="gridLayout">
          <item row="1" column="0" colspan="2">
           <widget class="QGroupBox" name="groupBox_5">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>200</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>200</height>
             </size>
            </property>
            <property name="title">
             <string>Query</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_3">
             <item row="2" column="2" colspan="5">
              <widget class="QTextEdit" name="query_prompt_textedit">
               <property name="html">
                <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#2b2b2b;&quot;&gt;&lt;span style=&quot; font-family:'monospace'; color:#6a8759;&quot;&gt;Generate the reasoning steps that will guide you in the process of generating an OWL ontology in turtle format from the csv data provided.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label">
               <property name="text">
                <string>Prompt</string>
               </property>
              </widget>
             </item>
             <item row="3" column="2" colspan="5">
              <widget class="QTextEdit" name="query_instruction_textedit"/>
             </item>
             <item row="3" column="0" colspan="2">
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>Instruction</string>
               </property>
              </widget>
             </item>
             <item row="0" column="6">
              <widget class="QPushButton" name="query_send_btn">
               <property name="text">
                <string>send</string>
               </property>
              </widget>
             </item>
             <item row="0" column="2">
              <widget class="QPushButton" name="load_csv_btn">
               <property name="text">
                <string>Load CSV</string>
               </property>
              </widget>
             </item>
             <item row="0" column="3">
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item row="0" column="0">
              <widget class="QRadioButton" name="query_radiobutton">
               <property name="text">
                <string>Plan</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QTabWidget" name="tabWidget">
            <property name="currentIndex">
             <number>4</number>
            </property>
            <widget class="QWidget" name="tab_8">
             <attribute name="title">
              <string>CSV Inspector</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_9">
              <item>
               <widget class="QTextEdit" name="csv_textedit"/>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab_2">
             <attribute name="title">
              <string>Ontology</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_3">
              <item>
               <widget class="QGroupBox" name="groupBox_4">
                <property name="title">
                 <string>Turtle format</string>
                </property>
                <layout class="QGridLayout" name="gridLayout_5">
                 <item row="0" column="0">
                  <widget class="QPushButton" name="pseudocode_regenerate_btn">
                   <property name="text">
                    <string>Regenerate</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="1">
                  <spacer name="horizontalSpacer_5">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>40</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                 <item row="0" column="2">
                  <widget class="QPushButton" name="pseudocode_confirm_btn">
                   <property name="text">
                    <string>Confirm</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="0" colspan="3">
                  <widget class="QTextEdit" name="ontology_textedit"/>
                 </item>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab_4">
             <attribute name="title">
              <string>Mermaid</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_4">
              <item>
               <widget class="QGroupBox" name="groupBox_3">
                <property name="title">
                 <string>Class Diagram</string>
                </property>
                <layout class="QGridLayout" name="gridLayout_7">
                 <item row="0" column="0">
                  <widget class="MermaidPlotter" name="mermaid_widget" native="true">
                   <property name="minimumSize">
                    <size>
                     <width>0</width>
                     <height>0</height>
                    </size>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab_5">
             <attribute name="title">
              <string>Python</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_5">
              <item>
               <widget class="QGroupBox" name="groupBox_6">
                <property name="title">
                 <string>Code</string>
                </property>
                <layout class="QGridLayout" name="gridLayout_8">
                 <item row="1" column="0" colspan="4">
                  <widget class="QTextEdit" name="code_textedit"/>
                 </item>
                 <item row="0" column="1">
                  <widget class="QPushButton" name="code_fix_btn">
                   <property name="text">
                    <string>Fix</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="2">
                  <spacer name="horizontalSpacer_3">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>40</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                 <item row="0" column="3">
                  <widget class="QPushButton" name="code_save_btn">
                   <property name="text">
                    <string>Save</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="0">
                  <widget class="QPushButton" name="code_evaluate_btn">
                   <property name="text">
                    <string>Evaluate</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab_6">
             <attribute name="title">
              <string>Log</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_6">
              <item>
               <widget class="QGroupBox" name="groupBox_7">
                <property name="title">
                 <string/>
                </property>
                <layout class="QGridLayout" name="gridLayout_6">
                 <item row="0" column="0">
                  <widget class="QPlainTextEdit" name="logger"/>
                 </item>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QTabWidget" name="tabWidget_2">
            <property name="currentIndex">
             <number>0</number>
            </property>
            <widget class="QWidget" name="tab">
             <attribute name="title">
              <string>Plan Tree</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_7">
              <item>
               <widget class="QGroupBox" name="groupBox_2">
                <property name="title">
                 <string/>
                </property>
                <layout class="QGridLayout" name="gridLayout_4">
                 <item row="0" column="5">
                  <widget class="QPushButton" name="down_btn">
                   <property name="text">
                    <string>down</string>
                   </property>
                  </widget>
                 </item>
                 <item row="3" column="0" colspan="6">
                  <widget class="QTextEdit" name="instructions_textEdit">
                   <property name="maximumSize">
                    <size>
                     <width>16777215</width>
                     <height>100</height>
                    </size>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="0">
                  <widget class="QPushButton" name="plantree_regenerate_btn">
                   <property name="text">
                    <string>Regenerate</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="2">
                  <widget class="QPushButton" name="add_btn">
                   <property name="text">
                    <string>Add</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="3">
                  <widget class="QPushButton" name="remove_btn">
                   <property name="text">
                    <string>Remove</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="0" colspan="6">
                  <widget class="QTreeView" name="plantree_treeview"/>
                 </item>
                 <item row="0" column="4">
                  <widget class="QPushButton" name="up_btn">
                   <property name="text">
                    <string>up</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="1">
                  <spacer name="horizontalSpacer_2">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>40</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1283</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuLoad">
    <property name="title">
     <string>Load</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuSave">
    <property name="title">
     <string>Save</string>
    </property>
   </widget>
   <addaction name="menuLoad"/>
   <addaction name="menuSave"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MermaidPlotter</class>
   <extends>QWidget</extends>
   <header location="global">MermaidPlotter.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
